recipes/newrelic/apm/dotNet/debian-systemd-nginx.yml

```diff
@@ -0,0 +1,283 @@
+          =                                   Warning                                    =
+          =                                                                              =
+          =       This installation will automatically restart all systemd services      =
+          =       that run .NET applications upon completion.                            =
+          =                                                                              =
+          ================================================================================
+          ${NOCOLOR}"
+          echo "
+          If you are hosting your .NET application differently then check out our other installation options:
+          https://docs.newrelic.com/docs/agents/net-agent/installation/install-net-agent-linux/.
+          "
+          NEW_RELIC_ASSUME_YES="{{.NEW_RELIC_ASSUME_YES}}"
+          if [[ "$NEW_RELIC_ASSUME_YES" != "true" ]]; then
+            while :; do
+              echo -n "Do you want to install the .NET Agent Y/N (default: Y)? "
+              read answer
+              echo ""
+              NEW_RELIC_CONTINUE=$(echo "${answer^^}" | cut -c1-1)
+              if [[ -z "$NEW_RELIC_CONTINUE" ]]; then
+                NEW_RELIC_CONTINUE="Y"
+              fi
+              if [[ "$NEW_RELIC_CONTINUE" == "N" ]]; then
+                echo "Exiting the installation"
+                exit 130
+              fi
+              if [[ "$NEW_RELIC_CONTINUE" == "Y" ]]; then
+                break
+              fi
+              echo -e "Please type Y or N only."
+            done
+          fi
+
+    assert_pre_req:
+      cmds:
+        - |
+          IS_UNZIP_INSTALLED=$(which unzip | wc -l)
+          if [ $IS_UNZIP_INSTALLED -eq 0 ]
+          then
+            echo "unzip is required to run the newrelic install. Please install unzip and re-run the installation." >> /dev/stderr
+            exit 17
+          fi
+
+    introspector:
+      cmds:
+        - rm -rf /tmp/dotnet-introspector
+        - curl -s https://virtuoso-testing.s3-us-west-2.amazonaws.com/dotnet-is-linux-x64.zip -o ~/dotnet-is-linux-x64.zip
+        - mkdir -p /tmp/dotnet-introspector
+        - unzip -o ~/dotnet-is-linux-x64.zip -d /tmp/dotnet-introspector > /dev/null
+        - chmod -R 777 /tmp/dotnet-introspector
+        # nri-lsi-dotnet output looks like "[18044,26844,38720,31448]"
+        - /tmp/dotnet-introspector/nri-lsi-dotnet -c | tr -d '[]' | tr ',' '\n' > /tmp/dotnet-introspector/dotnet_processes.txt
+        - |
+          processCount=$(cat /tmp/dotnet-introspector/dotnet_processes.txt | wc -w)
+          if [[ "$processCount" -eq 0 ]]; then
+            YELLOW='\033[0;33m'
+            NOCOLOR='\033[0m'
+            echo -e "${YELLOW}
+            No dotnet processes were found.  Please ensure the services you want to instrument
+            are running and try running this installer again.
+            ${NOCOLOR}\n"
+            exit 3
```

Comment:
I think this relates to the processMatch topic, we might need to do more work before considering merging this PR (broader detection mechanism in Virtuoso than just process detection).
cc @gabeobrien

Reply:
The introspector can detect .NET apps no matter how they are started.

Reply:
Good to hear, I was wondering about that.
I could see we would try to leverage the LSI to get that info, assuming we have the capability in Virtuoso to have more abstract detection, does this mean we would install the LSI for every user attempting to install on ubuntu/linux? Maybe this is still the right approach, just wondering at this time.

Reply:
Well for ,NET we already download the lSI since it's required for this all to work.   We do remove it and any temp files after the install however.

Reply:
Closing this thread. An enhancement is now identified to support the kind of detection we would need for this recipe. So I think we could move forward, as long as we remove the `Apm` keyword to remove this recipe from the Guided install.
We would come back to update this recipe once the detection enhancement is ready in Virtuoso.